jax/_src/tree.py

```diff
@@ -284,3 +286,45 @@ def unflatten(treedef: tree_util.PyTreeDef,
     - :func:`jax.tree.structure`
   """
   return tree_util.tree_unflatten(treedef, leaves)
+
+
+def field(*, static: bool = False, metadata: dict[str, Any] | None = None, **kwds) -> dataclasses.Field:
+  """Used to mark a jax.dataclass field as static.
+
+  See {func}`jax.tree.dataclass` for details.
+  """
+  return dataclasses.field(metadata=(metadata or {}) | {'static': static}, **kwds)
+
+
+@dataclass_transform(field_specifiers=(field, dataclasses.Field), frozen_default=True)
```

Comment:
The fact that `frozen` default to `True` could be surprising for `dataclasses.dataclass` users. Should we call that out in the docstring?

Also, would it be bad not to change the defaults wrt to the standard library `dataclass`?

Reply:
That's why you add the `replace` method :p  But srsly don't we want to encourage the frozen default for sanity w. all of JAX tracing?

Reply:
On Flax struct we recently added kwargs and users immediately figured out they could set `frozen=False` and they seem to enjoy it.